openmicroscopy/openmicroscopy
# Load server settings

This replace #4431 

This PR refactor the way omero properties are loaded into the webclient. It uses ``Iconfig.getClientConfigValues`` to load them all at once

New properties:
- `omero.client.ui.tree.orphans.enabled`

New alias properties, replacing deprecated in 5.3
- `omero.client.ui.menu.dropdown.leaders.label`
- `omero.client.ui.menu.dropdown.colleagues.label`
- `omero.client.ui.menu.dropdown.everyone.label`

-----

**To test:**
- [x] orphaned images:
   - set `omero.client.ui.tree.orphans.enabled false/False/true/True` and check if regular user cannot see Orphaned Images container in other user data tree. Admin and group owner should be able to access OI anytime. A user always should access its own OI
- [x] roi limit:
   - set `omero.client.viewer.roi_limit` and check if more then limit ROIs are not loaded at once
- [x] scripts:
   - set `omero.client.scripts_to_ignore` check if right scripts are hidden from the menu
- [x] download max size:
   - set `omero.client.download_as.max_size` and check if download as jpg/png/tiff is disabled above max pixel count and allowed below.
- [x] interpolate:
   - set `omero.client.viewer.interpolate_pixels false` check if checkbox in viewer is off
- [x] zoom level:
   - set `omero.client.viewer.initial_zoom_level 2` check if big image opens on that zoom level

also:

- [x] dropdown menu:
   - set `omero.client.ui.menu.dropdown.everyone.label` & `omero.client.ui.menu.dropdown.leaders.label` & `omero.client.ui.menu.dropdown.colleagues.label`
   - set `omero.client.ui.menu.dropdown.everyone.enabled false/true` & `omero.client.ui.menu.dropdown.leaders.enabled false/true` & `omero.client.ui.menu.dropdown.colleagues.enabled false/true`
- [x] dropdown menu backwards compatibility:
   - use develop latest or 5.2.1 release server (ask for help if you wish to redeploy eel/latest)
   - set `omero.client.ui.menu.dropdown.everyone` & `omero.client.ui.menu.dropdown.leaders` & `omero.client.ui.menu.dropdown.colleagues`
    - run webclient from this branch and check if that appear correctly on UI

cc: @dominikl Insight will need to take these to the account at some point
